### ISSUE TITLE
VSCode collapses all spaces in a liquid file schema on save

### ISSUE THREAD

**Issue by schellenbergk** [uicrooks/shopify-theme-lab#81]
### Description

VSCode collapses all spaces in a liquid file schema on save.  

<details open>
<summary>Additional Details</summary>
<br>
https://gifyu.com/image/D6wN
</details>

### What is Expected?

Properly formatted a Schema

### What is actually happening?

All spaces/new line characters removed

**Comment by arashshirani** [comment]
You're probably using prettier as your code formatter and that is causing the issue. a quick workaround is to add `prettier-ignore` comment before your schema.

```
{%- comment -%}<!-- prettier-ignore -->{%- endcomment -%}
{% schema %}
{
  ...
}
{% endschema %}
```

or the better alternative is to install the [Liquid](https://marketplace.visualstudio.com/items?itemName=sissel.shopify-liquid) extension and set that as your code formatter.

**Comment by sergejcodes** [comment]
I'm also using VS Code and don't have any problems. Most likely some formatter, like already mentioned, is causing the issue.

**Comment by schellenbergk** [comment]
Hmmm, none of that helped.  But I noticed by default when opening `.liquid` files the language mode detected is set to `HTML` not `liquid`, when I switched to `liquid` language (manually) then saving did not trigger auto format.  Format only worked properly when I run `Liquid: Format Document` command.

Is there a way to fix language detection in VSCode for liquid files and make the `format on save` work with it too?
